lib/collection/request-auth.js

```diff
@@ -74,8 +74,8 @@ _.assign(RequestAuth.prototype, /** @lends RequestAuth.prototype */ {
     update: function (options, type) {
         // update must have options
         if (!_.isObject(options)) { return; }
-        // validate type parameter and/or choose default from existing type.
-        if (!RequestAuth.isValidType(type)) { type = this.type; }
+        // validate type parameter and return in case type is not valid.
+        if (!RequestAuth.isValidType(type)) { return; }
```

Comment:
It should default to the current `this.type` if not provided but bail out if an invalid type is provided.

If this change is not failing any test, let's make sure there's a test for it.

Reply:
Fixed and added test.